components/dashboard/src/settings/Billing.tsx

```diff
@@ -6,13 +6,24 @@
 
 import { PageWithSettingsSubMenu } from "./PageWithSettingsSubMenu";
 import { BillingAccountSelector } from "../components/BillingAccountSelector";
+import UsageBasedBillingConfig from "../components/UsageBasedBillingConfig";
 
 export default function Billing() {
     return (
         <PageWithSettingsSubMenu title="Billing" subtitle="Usage-Based Billing.">
             <div>
                 <h3>Billing Account</h3>
                 <BillingAccountSelector />
+                <h3 className="mt-12">Usage-Based Billing</h3>
+                <UsageBasedBillingConfig
+                    userOrTeamId={""}
```

Comment:
I wonder whether this shouldn't be either `userId` or `teamId` 🤔 seems a bit awkward to have an ID that could be either for a user or a team. Also, the component will eventually need to know, because it needs to call either `createOrUpdateStripeCustomerForTeam` for teams (as currently implemented) or `createOrUpdateStripeCustomerForUser` for users (still to be implemented).

Reply:
Yes, I agree that the API of the component is not great here. 

My intention with the parent PR (https://github.com/gitpod-io/gitpod/pull/12732 where this prop was introduced) was to do a rough extraction of the components to allow them to be used in two places without duplicating them. As we work out exactly how this component is going to call the server API from the individual settings page we should be able to make the small changes needed to clean this up.

I've added a note to https://github.com/gitpod-io/gitpod/issues/12685 to revisit this.

Reply:
@jankeromnes are you happy to approve this, given that we'll revisit the API of the component as part of #12685?

Reply:
Now that we are a bit further along with the `server` side changes needed to support individual UBP, I've changed the prop from `userOrTeamId` to an `attributionId`.

I think this addresses the concern about the awkward prop name.